flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch6/ElasticsearchSink.java

```diff
@@ -86,6 +97,7 @@ private ElasticsearchSink(
 		private Map<String, String> bulkRequestsConfig = new HashMap<>();
 		private ActionRequestFailureHandler failureHandler = new NoOpFailureHandler();
 		private RestClientFactory restClientFactory = restClientBuilder -> {};
+		private IndexGenerator indexGenerator = new IndexGenerator(null, null, null);
```

Comment:
Does `IndexGenerator` is a must-have field? If yes, maybe we can just check it is set before `build()`. or we can create a default one before build()?